### ISSUE TITLE
AkkaBackend weird behavior to detect ConnectException 

### ISSUE THREAD

**Issue by aygalinc** [softwaremill/sttp#1058]
Hi,

Version use : 
val SttpVersion = "3.3.11"
          "com.softwaremill.retry" %% "retry" % "0.3.3",
          "com.softwaremill.sttp.client3" %% "core" % SttpVersion,
          "com.softwaremill.sttp.client3" %% "circe" % SttpVersion,
          "com.softwaremill.sttp.client3" %% "akka-http-backend" % SttpVersion

 I'm getting weird behavior with AkkaSttpBackend & retry when one of the service I rely is down.
I have draw a small test case :
```
"When constraint server is in error then we " should " get an error when asking for constraint" in {
val sttp = AkkaHttpBackend()

val request: () => Future[String] = () => {
  println("Launch some stuff")
  val startTime = System.currentTimeMillis()

  basicRequest
    .get(uri"https://fake.url:9000")
    .response(asString.getRight)
    .send(sttp)
    .map(_.body)
    .recoverWith {
      case error => {

        println(s"${System.currentTimeMillis() - startTime} millis")
        println(error.printStackTrace())
        Future.failed(error)

      }
    }
}

implicit val successForFuture: Success[String] = Success.always

recoverToSucceededIf[SttpClientException] {
  val response = retry.Pause(max = 3, delay = 20.milliseconds).apply {
    request
  }
  response
}
}

```
What I observe is inconsistent timeout for detecting the failure : 4366 millis - 200920 millis - 117760 millis - 138533 millis
So basically error detection occurs btw ' seconds to more than 3 minutes
I expect that the detection of this error would be quicker or at least take the Default 30 second of timeout no ?

**Comment by adamw** [comment]
Can you try a different backend (see https://sttp.softwaremill.com/en/latest/backends/future.html), so that we can more precisely localise the issue - is it in sttp, akka, configuration, retry, etc.

**Comment by aygalinc** [comment]
With okHttp : 125 millis - 3 millis - 2 millis - 2 millis

**Comment by adamw** [comment]
Now it would be great to check this using Akka HTTP client directly, so that we can further localise the issue

**Comment by aygalinc** [comment]
I get same strange behavior as in the initial test. SO Gulty seems to be the Akka Http client

**Comment by adamw** [comment]
@aygalinc if you've got a reproducing example using Akka HTTP client only, I think this should be moved to the https://github.com/akka/akka-http project

**Comment by adamw** [comment]
@aygalinc Although, I would also try to eliminate the retry library. Maybe it's some weird interaction between the two, who knows ;)

**Comment by aygalinc** [comment]
The issue id already open on akka side :)
